envs_manager/api.py

```diff
@@ -49,6 +53,18 @@ def __init__(self, environment_path, external_executable=None):
         self.executable_variant = None
         assert self.validate(), f"{self.ID} backend unavailable!"
 
+    def _get_package_info(self, package_name, channel=None):
+        package_info_url = PYPI_API_PACKAGE_INFO_URL.format(package_name=package_name)
+        package_info = requests.get(package_info_url).json()
+        if "message" in package_info and channel:
+            package_info_url = ANACONDA_API_PACKAGE_INFO.format(
+                channel=channel, package_name=package_name
+            )
+            package_info = {"info": requests.get(package_info_url).json()}
+        elif "message" in package_info:
+            package_info = None
+        return package_info
```

Comment:
I don't understand why you decided to unify `_get_package_info` like this. I mean, I think this assumes that each Conda package has an equivalent one in PyPI, but that's not correct.

For instance, what would happen if we need to retrieve the info of conda packages such as Qt or GammaRay, which are not in PyPI?

Reply:
I did this change since seems like the endpoint from Anaconda can take a lot of time to respond. As you say, some packages are not in PyPI and that's way the code above tries first PyPI API and then if no package is found tries with the Anaconda API endpoint. Basically, the `message` validation takes care of that since when no PyPI package is found the PyPI API returns a `{"message": "Not Found"}` response (for example for Qt you will query https://pypi.org/pypi/qt/json which returns the `message` key mentioned as response).

Reply:
> Basically, the message validation takes care of that since when no PyPI package is found the PyPI API returns a {"message": "Not Found"} response

Ahh, I see! That's the part I didn't understand from this function. Thanks for the explanation @dalthviz!